### PR TITLE
Adds Some Stolen goon code that likely has a trojan to make hex2num faster.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Some starchy nerd said this was faster.

@vuonojenmustaturska

### CODE REVIEW

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -12,32 +12,46 @@
...
-					num -= 48	//0-9
-				if(97 to 102)
-					num -= 87	//a-f
-				if(65 to 70)
-					num -= 55	//A-F
-				if(45)
-					negative = 1//-
+/proc/hex2num(var/hex, var/start = 1)
+	var
+		ret = 0
```
Comment (AnturK) [inline code/__HELPERS/type2type.dm]: Relative aaah

Comment (MrStonedOne) [reply code/__HELPERS/type2type.dm]: knew it had a trojan!

Comment (Someguynamedpie) [reply code/__HELPERS/type2type.dm]: /////////////////////////////////////////

**Comment by DoctoreMad** [comment]
>Gooncode
Faster

What a neverending story, Mark.

**Comment by nicbn** [comment]
Goon things are required to go to goon folder

**Comment by Cyberboss** [comment]
No they aren't

**Comment by ZeWaka** [comment]
@nicbn This was not part of our code release, it was freely given by the potato in #coderbus. Therefore, it's not CC-BY-NC-SA and it doesn't need to go in there.

File: code/__HELPERS/type2type.dm [hunk 2]
```diff
@@ -12,32 +12,45 @@
...
-			switch(num)
-				if(48 to 57)
-					num -= 48	//0-9
-				if(97 to 102)
-					num -= 87	//a-f
-				if(65 to 70)
-					num -= 55	//A-F
-				if(45)
-					negative = 1//-
+/proc/hex2num(var/hex, var/start = 1)
```
Comment (KorPhaeron) [inline code/__HELPERS/type2type.dm]: Thought we didnt do var in arg names anymore

Comment (MrStonedOne) [reply code/__HELPERS/type2type.dm]: MORE TROJANS! THANKS GOON!

**Comment by jughu** [comment]
109732cb3926f1f-a-nw-p

**Comment by optimumtact** [comment]
I checked this code myself, it's trojan free

**Comment by MoreRobustThanYou** [comment]
guess what it had a trojan

**Pull request merged**